diff --git a/node_modules/graphql-rabbitmq-subscriptions/dist/amqp-pubsub.js b/node_modules/graphql-rabbitmq-subscriptions/dist/amqp-pubsub.js
index f84a4a5..932536a 100644
--- a/node_modules/graphql-rabbitmq-subscriptions/dist/amqp-pubsub.js
+++ b/node_modules/graphql-rabbitmq-subscriptions/dist/amqp-pubsub.js
@@ -16,6 +16,7 @@ var AmqpPubSub = (function () {
         this.producer = new rabbitmq_pub_sub_1.RabbitMqPublisher(logger, factory);
         this.subscriptionMap = {};
         this.subsRefsMap = {};
+        this.unsubscribeChannelMap = {};
         this.currentSubscriptionId = 0;
     }
     AmqpPubSub.prototype.publish = function (trigger, payload) {
@@ -41,7 +42,7 @@ var AmqpPubSub = (function () {
                 _this.consumer.subscribe(triggerName, function (msg) { return _this.onMessage(triggerName, msg); })
                     .then(function (disposer) {
                     _this.subsRefsMap[triggerName] = (_this.subsRefsMap[triggerName] || []).concat([id]);
-                    _this.unsubscribeChannel = disposer;
+                    _this.unsubscribeChannelMap[id] = disposer;
                     return resolve(id);
                 }).catch(function (err) {
                     _this.logger.error(err, "failed to recieve message from queue '%s'", triggerName);
@@ -61,7 +62,7 @@ var AmqpPubSub = (function () {
         var newRefs;
         if (refs.length === 1) {
             newRefs = [];
-            this.unsubscribeChannel().then(function () {
+            this.unsubscribeChannelMap[subId]().then(function () {
                 _this.logger.trace("cancelled channel from subscribing to queue '%s'", triggerName);
             }).catch(function (err) {
                 _this.logger.error(err, "channel cancellation failed from queue '%j'", triggerName);
