diff --git a/node_modules/rabbitmq-pub-sub/dist/main.js b/node_modules/rabbitmq-pub-sub/dist/main.js
index 5b40d7e..75d6dd4 100644
--- a/node_modules/rabbitmq-pub-sub/dist/main.js
+++ b/node_modules/rabbitmq-pub-sub/dist/main.js
@@ -248,7 +248,7 @@ var RabbitMqConsumer = (function () {
     RabbitMqConsumer.prototype.getDLSettings = function () {
         return {
             durable: true,
-            autoDelete: false
+            autoDelete: true
         };
     };
     return RabbitMqConsumer;
@@ -293,7 +293,7 @@ var RabbitMqProducer = (function () {
     RabbitMqProducer.prototype.getQueueSettings = function (deadletterExchangeName) {
         return {
             durable: true,
-            autoDelete: false,
+            autoDelete: true,
             arguments: {
                 'x-dead-letter-exchange': deadletterExchangeName
             }
@@ -485,6 +485,7 @@ var RabbitMqSubscriber = (function () {
     };
     RabbitMqSubscriber.prototype.getQueueSettings = function (deadletterExchangeName) {
         return {
+            autoDelete: true,
             exclusive: true
         };
     };
